src/linker/Linker.Steps/UnreachableBlocksOptimizer.cs

```diff
@@ -64,54 +65,17 @@ public ProcessingNode (in ProcessingNode other, int newLastAttempStackVersion)
 		//   - Instruction instance - method has been processed and it has a constant return value (the value of the instruction)
 		// Note: ProcessedUnchangedSentinel is used as an optimization. running constant value analysis on a method is relatively expensive
 		// and so we delay it and only do it for methods where the value is asked for (or in case of changed methods upfront due to implementation detailds)
-		Dictionary<MethodDefinition, Instruction> processedMethods;
+		readonly Dictionary<MethodDefinition, Instruction> _processedMethods;
 		static readonly Instruction ProcessedUnchangedSentinel = Instruction.Create (OpCodes.Ldstr, "ProcessedUnchangedSentinel");
 		static readonly Instruction NonConstSentinel = Instruction.Create (OpCodes.Ldstr, "NonConstSentinel");
 
-		protected override void Process ()
+		public UnreachableBlocksOptimizer (LinkContext context)
 		{
-			var assemblies = Context.Annotations.GetAssemblies ().ToArray ();
+			_context = context;
 
-			processingStack = new LinkedList<ProcessingNode> ();
-			processingMethods = new Dictionary<MethodDefinition, LinkedListNode<ProcessingNode>> ();
-			processedMethods = new Dictionary<MethodDefinition, Instruction> ();
-
-			foreach (var assembly in assemblies) {
-				if (Annotations.GetAction (assembly) != AssemblyAction.Link)
-					continue;
-
-				ProcessMethods (assembly.MainModule.Types);
-			}
-		}
-
-		void ProcessMethods (Collection<TypeDefinition> types)
-		{
-			foreach (var type in types) {
-				if (type.IsInterface)
```

Comment:
Looks like we no longer check for interfaces - does it just mean that default interface methods can have branches removed?

Reply:
Good catch - and basically yes. I fixed it (unintentionally 😉 ). So I now added at least a basic test that branch removal works within default interface methods.